beacon-chain/types/active_state.go

```diff
@@ -228,8 +251,33 @@ func (a *ActiveState) CalculateNewActiveState(
 		}
 	}
 
+	_, proposerIndex, err := casper.ProposerShardAndIndex(
+		cState.ShardAndCommitteesForSlots(),
+		cState.LastStateRecalculationSlot(),
+		parentSlot)
```

Comment:
Hm. We have used `parentSlot` across our code base. I double checked it's `parentSlot` because of the committee from last slot determines who is the proposer of the current slot

Reply:
The wording in the spec is confusing, but I'm pretty sure that this should be `block.SlotNumber()`. 

> Let proposer_index be the validator index of the parent.slot % len(get_shards_and_committees_for_slot(crystallized_state, parent.slot)[0].committee)'th attester in get_shards_and_committees_for_slot(crystallized_state, parent.slot)[0]. Verify that an attestation from this validator is part of the first (ie. item 0 in the array) AttestationRecord object; **this attester can be considered to be the proposer of the parent block**.

If the above quote is what you were referring to, it describes how the parent block's proposer's attestation should be included in the current block. This logic in this chnageset involves the current block's proposer.

Reply:
The parentSlot comes from this line
>Let proposer_index be the validator index of the parent.slot % len(get_shards_and_committees_for_slot(crystallized_state, parent.slot)[0].committee)'th attester

You think it should be `current.slot % len(get_shards_and_committees_for_slot(crystallized_state, current.slot)[0].committee)'th attester`?

Reply:
Yeah, check out the line I emphasized in bold. That describes the proposer of the parent block. We want the proposer of the current block instead.

Reply:
super confusing!

Reply:
This PR makes clear the difference between the curblock proposer and parentblock proposer
https://github.com/ethereum/eth2.0-specs/pull/69